create_derivatives/routines.py

```diff
@@ -68,8 +69,62 @@ class JP2Maker:
 
 
 class PDFMaker:
-    # TO DO: make PDF derivates, compress, OCR
-    pass
+    """Creates concatenated PDF file from JP2 derivatives.
+
+    Creates PDF directory in bag's data directory, creates PDF, then compresses and OCRs the PDF
+
+    Returns:
+        A tuple containing human-readable message along with list of bag identifiers.
+        Exceptions are raised for errors along the way.
+
+    """
+
+    def run(self):
+        bags_with_pdfs = []
+        for bag in Bag.objects.filter(process_status=Bag.JPG2000):
+            jp2_files_dir = Path(bag.bag_path, "data", "JP2")
```

Comment:
There are a couple of spots here where you are passing around a Path instance but you might need to coerce it to a string before using it somewhere.